pallets/xcm-transactor/src/lib.rs

```diff
@@ -219,7 +225,21 @@ pub mod pallet {
 			IndexToAccount::<T>::insert(&index, who.clone());
 
 			// Deposit event
-			Self::deposit_event(Event::<T>::RegisterdDerivative(who, index));
+			Self::deposit_event(Event::<T>::RegisteredDerivative(who, index));
+
+			Ok(())
+		}
+
+		#[pallet::weight(T::WeightInfo::deregister())]
+		/// De-Register a derivative index.
```

Comment:
"derivative" could use a bit more context here -- is this related to sovereign accounts?

Reply:
yes, an account derived from the sovereign. WIll rpovide better description